### ISSUE TITLE
Removal of File_gump, Exult 1.10 or SDL 3

### ISSUE THREAD

**Issue by Dragon-Baroque** [exult/exult#532]
Can we discard the old Save / Load `File_gump` ?

It looks like it was for Game Load and Save, but testing it as of `master` over SDL 2, it throws invalid Shapes, is not responsive to the keyboard, and its only working button is `Quit` to leave Exult entirely.

It is also the only Gump yet to use the `unicode` data variable which looks to me as seriously wrong ( UCS2 perhaps, this might explain why it does not react to the keyboard ), whereas the replacement `NewFile_gump` and also the `Notebook_gump` use SDL TextInput instead.

In conclusion, I would like to discard that Gump and the references to it ( old_file_gump ), and remove the `unicode` , preferably for 1.10, otherwise for SDL 3, since it drags down the keyboard rework for SDL 3.

**Comment by DominusExult** [comment]
I never used it except for testing, probably *decades* ago, I have no problem removing it.
Would make http://exult.info/faq.php#disable_improvements a lot shorter :)

**Comment by wench** [comment]
I have no problems with this. We could add a mode to make the newfile gump to behave more like the old one if people really don't want to use the new one.

**Comment by Dragon-Baroque** [comment]
Thank you. 

On the other hand, `unicode` is not so easy to remove. It is in fact used as some kind of interface between the TextInput and the KeyDown events. 

So that neither the `NewFile_gump` nor the `Notebook_gump` allow me to enter accented characters. Can you enter accented text in the Notebook ( master, SDL 2 ) ?

> We could add a mode to make the newfile gump to behave more like the old one if people really don't want to use the new one.

Could you quickly tell how differently the two file gumps behave ?

**Comment by wench** [comment]
The new one allows unlimited saves and shows extended meta data and a mini screenshot while the old one limits to 10 saves and the old one shows nothing except the name of the save game.
The old one is also supposed to have buttons for audio options that the new one does not have.

**Comment by Dragon-Baroque** [comment]
> The new one allows unlimited saves and shows extended meta data and a mini screenshot while the old one limits to 10 saves and the old one shows nothing except the name of the save game.

So proposing the old behaviour is just making some _feature_ conditional. Fine.

> The old one is also supposed to have buttons for audio options that the new one does not have.

It displayed them, yes, but they did not have any effect.

**Comment by DominusExult** [comment]
> So that neither the NewFile_gump nor the Notebook_gump allow me to enter accented characters. Can you enter accented text in the Notebook ( master, SDL 2 ) ?

Well... fonts.vga is not exactly unicode. However we could bundle the fonts.vga  [Alternate fonts based on Kanto, enhanced by Sir John and Dominus for all official releases](http://exult.info/snapshots/fonts_replacement.zip) that @ungesundes-halbwissen adapted. much more characters than lock stock fonts.vga.
And we could give the option to replace the game fonts with the much more readable one, without having to place this in the patch folder.

**Comment by wench** [comment]
Yeah accented characters need specific font support and only the non english versions have a default fonts.vga that even has accented characters but we'd need to covert from unicode to whateve character sets those game versions use

Cheat screen input obviously has the same problem for string input. I wonder what do we do about your name in character creation for non english versions.

**Comment by Dragon-Baroque** [comment]
> Yeah accented characters need specific font support.

I keep forgetting that. Sorry.

> The new one allows unlimited saves and shows extended meta data and a mini screenshot while the old one limits to 10 saves and the old one shows nothing except the name of the save game.
> The old one is also supposed to have buttons for audio options that the new one does not have.

With a rework of the Unicode support in SDL3, the old one works again, and I have understood that the Sound buttons were not supposed to bring the Audio gumps but to directly enable / disable their type of audio and that this works.

So what about, for the users who prefer the old style save / load, putting an "Old" button in the New Save / Load window, and a "New" button into the Old Save / Load window ? Further away we could add a configuration option that remembers whether the last Save / Load window was the Old one or the New one ?

**Comment by DominusExult** [comment]
If you want to add an easier way to use the old save gump, I‘d say add an option to the cfg and a GUI button in the Game Display gump.
But IMO that‘s way too much work for an option no one seemingly uses.
Seemingly because even though there is a way described in the FAQ to use it, no one has asked for help with it in years (and since no one RTFM this is the tell for me).

**Comment by Dragon-Baroque** [comment]
> But IMO that‘s way too much work for an option no one seemingly uses.

OK then.
I can easily make it work again - this is commenting out the unicode related lines. Like the new Save/Load, actually. Very small patch.

And I can - proposal - attach it to Shift-S. No exchange button, no config option. Just a one added line to data/ bg + si / defaultkeys. Plus I would like to rename it from old_file_gump to old_save_restore, this is cosmetic.

Grand total : 4 files changed, 4 insertions(+), 6 deletions(-). Do you agree ?

**Comment by Dragon-Baroque** [comment]
With the commit #https://github.com/exult/exult/commit/24d15455b082a238f25cda0fa20566f5dccc3505, the Old style Save / Load works again, and neither  the Old style nor the New style Save / Load do throw `nullptr SHAPE!!!` errors or crashes.

At this point, I suggest to stand there, to leave the Old style Save / Load in the source tree, but not to bind it to any key.
And to close the current issue.
